packages/services/document-service/src/main/java/software/uncharted/terarium/documentserver/resources/xdd/DictionariesResource.java

```diff
@@ -22,7 +28,18 @@ public class DictionariesResource {
 	@Produces(MediaType.APPLICATION_JSON)
 	@Tag(name = "Get available XDD dictionaries")
 	@Path("/dictionaries")
+	@CacheResult(cacheName = CACHE_NAME)
 	public XDDResponse<XDDDictionariesResponseOK> getAvailableDictionaries(@QueryParam("all") @DefaultValue("") String all) {
-		return proxy.getAvailableDictionaries(all);
+		try {
+			XDDResponse<XDDDictionariesResponseOK> response = proxy.getAvailableDictionaries(all);
+		} catch (RuntimeException e) {
+			log.error("Unable to get dictionaries. Invalidating cache.", e);
```

Comment:
I don't think that's what you want here. I'm guessing if this behaves like other caches that it will not cache the result if the wrapped method throws an exception.